daffodil-udf/src/test/scala/org/sbadudfs/udfpexceptions2/StringFunctions/StringFunctionsProvider.scala

```diff
@@ -32,7 +32,7 @@ class StringFunctionsProvider extends UserDefinedFunctionProvider {
   ) extends Exception(message, cause)
 
   throw new CustomException("UDFP Error!")
-  override def getUserDefinedFunctionClasses = {
+  override def getUserDefinedFunctionClasses: Array[Class[?]] = {
```

Comment:
I don't know that we would want to do it here, but Scala 3 has union types that might be useful for something like these cases where we have Array[Class[?]]

Reply:
In this case `getUserDefinedFunctionClasses` is the implementation of an Java API function that returns a `Class<?>[]`, so it's just an array of Class'es with no restriction on the type of class that it can return. So you're right that we can't use them here. I imagine there are places in our code base where they could be useful if we ever switch to Scala 3.